### PR TITLE
fix(app-activity): Update app activity to match comment/task

### PR DESCRIPTION
**Pull request by jstoffan**
I took the opportunity to port it to the Media component and to update the styles to the SUIT naming conventions. We should consider pulling out and reusing the common pieces of AppActivity and Comment to avoid them drifting apart again in the future.

activity-new-1
activity-new-2

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js [hunk 1]
```diff
@@ -68,54 +65,107 @@ function mapActivityNodes(node: HTMLLinkElement): React.Node {
...
+                <Media.Body className="bcs-AppActivity-body">
+                    {isMenuVisible && (
+                        <TetherComponent
+                            attachment="top right"
+                            className="bcs-AppActivity-confirm"
+                            constraints={[{ to: 'scrollParent', attachment: 'together' }]}
+                            targetAttachment="bottom right"
+                        >
+                            <Media.Menu isDisabled={isConfirmingDelete}>
+                                {canDelete && (
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js]: Is this `canDelete` necessary if `isMenuVisible` already takes that into account?

File: src/components/media/Media.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import './Media.scss';
 
 type Props = {
     /** Component to use as outermost element, e.g., 'li' */
-    as: React.ElementType,
+    as?: React.ElementType,
```
Comment (priyajeet) [inline src/components/media/Media.js]: Its defaulted below, shouldn't need ?

File: src/components/media/MediaFigure.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import './Media.scss';
 
 type Props = {
     /** Component to use as outermost element, e.g., 'div' */
-    as: React.ElementType,
+    as?: React.ElementType,
```
Comment (priyajeet) [inline src/components/media/MediaFigure.js]: same

File: src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js [hunk 4]
```diff
@@ -68,54 +65,107 @@ function mapActivityNodes(node: HTMLLinkElement): React.Node {
...
+                                    <MenuItem onClick={this.handleDeleteClick}>
+                                        <IconTrash color={bdlGray80} />
+                                        <FormattedMessage {...deleteMessages.deleteLabel} />
+                                    </MenuItem>
+                                )}
+                            </Media.Menu>
+
+                            {isConfirmingDelete && (
+                                <CommentDeleteConfirmation
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js]: Should we rename this for the future? `CommentDeleteConfirmation` seems misleading

File: src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js [hunk 5]
```diff
@@ -4,22 +4,20 @@
...
-
 type Props = {
     isOpen: boolean,
+    message: MessageDescriptor,
     onDeleteCancel: Function,
     onDeleteConfirm: Function,
-    type: typeof COMMENT_TYPE_DEFAULT | typeof COMMENT_TYPE_TASK,
 };
 
 class CommentDeleteConfirmation extends React.Component<Props> {
```
Comment (priyajeet) [inline src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js]: Sateless, maybe make it at least PureComponent for now?

File: src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js (hunk 4 above)
Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js]: I think that can be deferred to a future refactor where we can put some more effort into sharing behavior between activity types.

File: src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js (hunk 5 above)
Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js]: Maybe, but I don't plan to refactor this component further as part of this pull request.

File: src/components/media/Media.js (hunk 2 above)
Comment (jstoffan) [reply src/components/media/Media.js]: Yeah, will fix. Was having some local flow server issues, I think.

File: src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js (hunk 1 above)
Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js]: It's following the same convention as most of our other menus, but I can remove it.

**Comment by alexkrolick** [comment]
> We should consider pulling out and reusing the common pieces of AppActivity and Comment to avoid them drifting apart again in the future.

Yeah, such as "FeedTimestamp" and maybe the title area. Lineheight is also very specific.

**Pull request merged**